stubs/gdb/gdb/__init__.pyi

```diff
@@ -23,7 +23,12 @@ STDOUT: int
 STDERR: int
 STDLOG: int
 
-def execute(command: str, from_tty: bool = ..., to_string: bool = ...) -> str | None: ...
+@overload
+def execute(command: str, from_tty: bool = False, to_string: Literal[True] = True) -> str: ...
+@overload
+def execute(command: str, *, to_string: Literal[False]) -> None: ...
+@overload
+def execute(command: str, from_tty: bool = False, to_string: bool = False) -> str | None: ...
```

Comment:
The default is `False`, so the overloads should be:

```suggestion
@overload
def execute(command: str, *, to_string: Literal[True]) -> str: ...
@overload
def execute(command: str, from_tty: bool, to_string: Literal[True]) -> str: ...
@overload
def execute(command: str, from_tty: bool = False, to_string: bool = False) -> str | None: ...
```

That said, doesn't it always return `None` if `to_string` is `False` (or unset)? In this case, the last overload could use `Literal[False]` and return just `None`.

Reply:
Thank you! I thought the last overload is required when passing a variable to `to_string` so you don't know statically if it's True or False? Maybe I'm misunderstanding.

Reply:
I can never remember this, but I think type checkers should be able to combine overloads in that case. But it's also possible that we need a fourth overload, i.e. one with `Literal[False]` and one with `bool`.

Reply:
I've tested with 
```python
from __future__ import annotations
from typing import overload
from typing_extensions import Literal


@overload
def execute(command: str, from_tty: bool = False, to_string: Literal[False] = False) -> None: ...
@overload
def execute(command: str, *, to_string: Literal[True]) -> str: ...
@overload
def execute(command: str, from_tty: bool, to_string: Literal[True]) -> str: ...
@overload
def execute(command: str, from_tty: bool = False, to_string: bool = False) -> str | None: ...

reveal_type(execute("bt"))
reveal_type(execute("bt", to_string=False))
reveal_type(execute("bt", from_tty=True, to_string=False))
reveal_type(execute("bt", from_tty=False, to_string=False))
reveal_type(execute("bt", from_tty=True))
reveal_type(execute("bt", from_tty=False))
reveal_type(execute("bt", to_string=True))
reveal_type(execute("bt", from_tty=True, to_string=True))
reveal_type(execute("bt", from_tty=False, to_string=True))

def some_function(thing: bool) -> None:
    reveal_type(execute("bt", to_string=thing))
    reveal_type(execute("bt", from_tty=True, to_string=thing))
    reveal_type(execute("bt", from_tty=False, to_string=thing))
```

and mypy prints the expected types
```
$ mypy /tmp/execute.pyi
/tmp/execute.pyi:15: note: Revealed type is "None"
/tmp/execute.pyi:16: note: Revealed type is "None"
/tmp/execute.pyi:17: note: Revealed type is "None"
/tmp/execute.pyi:18: note: Revealed type is "None"
/tmp/execute.pyi:19: note: Revealed type is "None"
/tmp/execute.pyi:20: note: Revealed type is "None"
/tmp/execute.pyi:21: note: Revealed type is "builtins.str"
/tmp/execute.pyi:22: note: Revealed type is "builtins.str"
/tmp/execute.pyi:23: note: Revealed type is "builtins.str"
/tmp/execute.pyi:26: note: Revealed type is "Union[builtins.str, None]"
/tmp/execute.pyi:27: note: Revealed type is "Union[builtins.str, None]"
/tmp/execute.pyi:28: note: Revealed type is "Union[builtins.str, None]"
Success: no issues found in 1 source file
```

So both overloads, one with Literal[False] and one with bool, are required it seems.